### DISCUSSION TITLE
adding a price-per-key limit during auto upgrade/buy cards

### DISCUSSION THREAD

**Discussion by omidRMR** [masterking32/MCF_HamsterKombat#9] [Ideas]
![image](https://github.com/user-attachments/assets/c7d886e5-db0f-42e5-bcee-a21a607fc869)
 I personally prefer not to buy/upgrade any card at any price.  I edited the code in an unprofessional way.  It is working for me. 
Perhaps others also want this feature.
![image](https://github.com/user-attachments/assets/5d4c2bdb-e1be-4849-9e05-87e0f775e290)

**Comment by Fy0urM** [comment]
In the original code, a card is not bought if its profit coefficient is greater than 200, the coefficient is calculated as [card price]/[card level pph].
https://github.com/masterking32/MCF_HamsterKombat/blob/main/FarmBot%2Fcore%2FCards.py#L42

You literally added the same check in a different place.

**Reply by omidRMR** [reply]
You're right, I didn't notice that line.

**Comment by Fy0urM** [comment]
The only difference is that in the original code the check occurs in the purchase cycle, while the cards are sorted by benefit from higher to lower and if the card coefficient is higher than allowed, the purchase cycle is interrupted, this will be finalized a little later

**Reply by omidRMR** [reply]
Yes, my change is quite useless. :-).

**Reply by Fy0urM** [reply]
Not really, you check the coefficient when you form cards, and in this case all cards that fit the coefficient restrictions will be bought. In the original code, if the first card in the array doesn't fit, the buying cycle is interrupted and the rest of the cards are not bought. I will fix this a little later. Now I was left without light, I was just modifying the buying algorithm).

**Reply by omidRMR** [reply]
To be honest, the idea of FarmBot is great, but it made all the steps too complicated to follow through the code (at least for me). Before releasing the FarmBot, you said (I don't remember where, maybe in some pull requests) that you would release an update for the old bot. But unfortunately, Mr. Amin decided to archive the old project. After, hanging around the FarmBot for days, I think I should switch back to the old project and modify the old code based on your valuable work in this project, piece by piece. 
Thank you for all you have done and everything you will do.

**Reply by Fy0urM** [reply]
Most of the work was done by masterking, I only supplemented it a bit. Yes, in some places the old bot was better, but I still like the new one better, even though it is more difficult to understand, because the main code belongs to masterking as well.
